nengo/builder/connection.py

```diff
@@ -53,6 +57,50 @@ def build_linear_system(model, conn, rng):
     return eval_points, activities, targets
 
 
+def build_decoders(model, conn, transform, rng):
+    encoders = model.params[conn.pre_obj].encoders
+    gain = model.params[conn.pre_obj].gain
+    bias = model.params[conn.pre_obj].bias
+
+    eval_points = get_eval_points(model, conn, rng)
+    targets = get_targets(model, conn, eval_points)
+
+    x = np.dot(eval_points, encoders.T / conn.pre_obj.radius)
+    E = None
+    if conn.solver.weights:
+        E = model.params[conn.post_obj].scaled_encoders.T
+        # account for transform
+        targets = np.dot(targets, transform.T)
+        transform = np.array(1., dtype=np.float64)
+
+    try:
+        decoders, solver_info = model.decoder_cache.wrap_solver(
+            solve_for_decoders)(
+                conn.solver, conn.pre_obj.neuron_type, gain, bias, x, targets,
+                rng=rng, E=E)
+    except ZeroActivityError:
+        raise ZeroActivityError(
+            "Building %s: 'activities' matrix is all zero for %s. "
+            "This is because no evaluation points fall in the firing "
+            "ranges of any neurons." % (conn, conn.pre_obj))
```

Comment:
[We're currently raising an exception](https://github.com/nengo/nengo/pull/677/files#diff-d88c987696c16673f33089794ccd0bbfR51) so this just keeps the existing behaviour and just names the exception so that it is possible to catch it without catching all `RuntimeError`.

I think, it should at least give a warning. But I can't imagine a case where a all zero activities matrix would not indicate a definitely broken model.